llvm/lib/Target/AMDGPU/AMDGPUInsertSingleUseVDST.cpp

```diff
@@ -132,6 +133,14 @@ class AMDGPUInsertSingleUseVDST : public MachineFunctionPass {
 
   AMDGPUInsertSingleUseVDST() : MachineFunctionPass(ID) {}
 
+  static bool isValidProducerInst(const MachineInstr &MI) {
```

Comment:
Probably simpler to inline this function into its only call site. Then we would not have a mix of "isValid" and "isInvalid" functions, which I find a little confusing.

Reply:
Done